contrib/gofiber/fiber.v2/option.go

```diff
@@ -75,13 +77,11 @@ func WithAnalyticsRate(rate float64) Option {
 	}
 }
 
-// WithStatusCheck allow setting of a function to tell whether a status code is an error
-func WithStatusCheck(fn func(statusCode int) bool) Option {
+// WithStatusCheck is deprecated. Please use globalconfig to set client/server
+// status codes. WithStatusCheck allows setting of a function to tell whether a
+// status code is an error.
+func WithStatusCheck(b bool) Option {
 	return func(cfg *config) {
-		cfg.isStatusError = fn
+		cfg.isStatusError = nil
 	}
 }
-
-func isServerError(statusCode int) bool {
-	return statusCode >= 500 && statusCode < 600
-}
```

Comment:
We can't break this function. It will cause customer code to fail to compile. 

Marking it deprecated is sufficient. Let's also not mention `globalconfig` in the documentation since that's not exposed to customers. We should have them use `WithHTTPServerErrorStatuses` instead.